main.tf

```
@@ -159,16 +159,18 @@ resource "aws_lambda_function" "default" {
   architectures                  = [var.architecture]
   code_signing_config_arn        = var.code_signing_config_arn
   description                    = var.description
-  filename                       = var.s3_bucket == null ? local.filename : null
+  filename                       = var.s3_bucket == null ? var.image_uri == null ? local.filename : null : null
```

```suggestion
  filename                       = var.s3_bucket == null || var.image_uri == null ? local.filename : null
```

We can skip the nested operator here.

---

Change applied